pkg/engine/plan.go

```diff
@@ -242,13 +243,17 @@ func (planResult *planResult) Close() error {
 }
 
 // printPlan prints the plan's result to the plan's Options.Events stream.
-func printPlan(ctx *Context, planResult *planResult, dryRun bool) (ResourceChanges, error) {
+func printPlan(ctx *Context, planResult *planResult, dryRun bool) (ResourceChanges, result.Result) {
 	planResult.Options.Events.preludeEvent(dryRun, planResult.Ctx.Update.GetTarget().Config)
 
 	// Walk the plan's steps and and pretty-print them out.
 	actions := newPlanActions(planResult.Options)
-	if err := planResult.Walk(ctx, actions, true); err != nil {
-		return nil, errors.New("an error occurred while advancing the preview")
+	if res := planResult.Walk(ctx, actions, true); res != nil {
+		if res.IsBail() {
+			return nil, res
+		}
+
+		return nil, result.Error("an error occurred while advancing the preview")
```

Comment:
unfortunate.  it seems like we've lost the original err information (this was a problem before).

Reply:
I think that's OK - we should have already printed it, right?

Reply:
Oh, nevermind - in this case, we won't have printed it. Could you instead return `res` verbatim without creating the new error?